From 1d449044caf2224da5e55d0de51459137feb0807 Mon Sep 17 00:00:00 2001
From: Pavel Moravec <pmoravec@redhat.com>
Date: Sun, 25 May 2025 11:46:34 +0200
Subject: [PATCH] [cleaner] Use "hostname -f" in HostnamePrepper

HostnamePrepper should always be fed by FQDN to ensure domain names are
recognized properly.

This is important esp. when:
- /etc/hosts is empty
- "hostname" contains shortname only
- "hostname -f" contains FQDN

Resolves: #4022
Closes: #4026

Signed-off-by: Pavel Moravec <pmoravec@redhat.com>
---
 sos/cleaner/preppers/hostname.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sos/cleaner/preppers/hostname.py b/sos/cleaner/preppers/hostname.py
index 0812597e..b20f5679 100644
--- a/sos/cleaner/preppers/hostname.py
+++ b/sos/cleaner/preppers/hostname.py
@@ -29,7 +29,7 @@ class HostnamePrepper(SoSPrepper):
         items = []
         _file = 'hostname'
         if archive.is_sos:
-            _file = 'sos_commands/host/hostname'
+            _file = 'sos_commands/host/hostname_-f'
         elif archive.is_insights:
             _file = 'data/insights_commands/hostname_-f'
 
-- 
2.49.0

